lib/fuel-auth.js

```diff
@@ -20,19 +20,26 @@ module.exports = class FuelAuth {
 			if (typeof options.clientId !== 'string' || typeof options.clientSecret !== 'string') {
 				throw new Error('clientId or clientSecret must be strings');
 			}
+
+			if(options.apiVersion === 2 && !options.tenant){
+				throw new Error('Version 2 APIs require that you specify a tenant');
+			}
 		} else {
 			throw new Error('options are required. see readme.');
 		}
 
 		// set required values
 		this.accessToken = options.accessToken;
-		this.authUrl = options.authUrl || 'https://auth.exacttargetapis.com/v1/requestToken';
+		this.authUrl = options.authUrl || (options.version === 2 ? `https://${options.tenant}.auth.marketingcloudapis.com/v2/token` : 'https://auth.exacttargetapis.com/v1/requestToken');
 		this.clientId = options.clientId;
 		this.clientSecret = options.clientSecret;
 		this.expiration = null;
 		this.refreshToken = options.refreshToken;
 		this.scope = options.scope;
+		this.tenant = options.tenant;
```

Comment:
What is the purpose of `authUrl` as an optional parameter? From what I understand the conditional that handle it being missing are robust to cover the cases. I imagine the use is how `v1` works for legacy packages, as such I think now with the `v2` support the documentation for this parameter should really reflect the goal of this parameter/variable.

The addition of adding tenant suggests that the way to form the `authUrl` is to analyze what options are being passed in the constructor and based on that interpretation form the correct `authUrl` to usage. 

I am a fan of this approach, and I see no real benefit in allowing `authUrl` to even be passed. But this is more of a question of clarity than a request for change.

Reply:
authUrl has been used previously as there was the chance it would change. I think for all V2 we dont have to worry about it , but again in an attempt to avoid breaking changes, we can still support both (even if I agree, the actual usage shouldn't really exist).